### PR TITLE
Add advertised endpoints.

### PR DESCRIPTION
**Pull request by neunhoef**
Fixes arangodb/team-clifton-planning#32

### CODE REVIEW

**Comment by neunhoef** [comment]
Compiles now. No guarantees for correctness.

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/843/

File: arangod/Cluster/ClusterFeature.cpp [hunk 1]
```diff
@@ -143,21 +143,41 @@ void ClusterFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
...
       << "must at least specify one endpoint in --cluster.agency-endpoint";
     FATAL_ERROR_EXIT();
   }
+  
+  // validate --cluster.my-address
+  if (_myEndpoint.empty()) {
+    LOG_TOPIC(FATAL, arangodb::Logger::CLUSTER) << "unable to determine internal address for server '"
+    << ServerState::instance()->getId()
+    << "'. Please specify --cluster.my-address or configure the "
+    "address for this server in the agency.";
```
Comment (neunhoef) [inline arangod/Cluster/ClusterFeature.cpp]: We should adjust this error message, since we no longer support to configure the address via the agency.

**Comment by kvahed** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1092/

**Pull request merged**